test/arrayops.jl

```diff
@@ -2651,6 +2651,14 @@ end
 # Throws ArgumentError for negative dimensions in Array
 @test_throws ArgumentError fill('a', -10)
 
+@testset "setindex" begin
+    ==ₜ(_, _) = false
+    ==ₜ(x::T, y::T) where T = x == y
+
+    @test Base.setindex(Int[1, 2], 3.0, 2) ==ₜ [1.0, 3.0]
```

Comment:
```suggestion
    @test Base.setindex([1, 2], 3.0, 2) ==ₜ [1.0, 3.0]
```

Reply:
I think it is good to have `Int` explicit, since the test is about change of eltype.

Reply:
What's the point since that doesn't make any difference?

Reply:
emphasis